src/mlpack/core/optimizers/sgd/sgd.hpp

```diff
@@ -72,7 +75,7 @@ namespace optimization {
  * @tparam DecomposableFunctionType Decomposable objective function type to be
  *     minimized.
  */
-template<typename DecomposableFunctionType>
+template<typename DecomposableFunctionType, typename UpdatePolicyType>
```

Comment:
In favor of backward compatibility, I would go with a default for UpdatePolicyType.